daffodil-core/src/main/scala/org/apache/daffodil/dsom/SchemaComponent.scala

```diff
@@ -169,36 +156,19 @@ trait SchemaComponent
   /**
    * Elements only e.g., /foo/ex:bar
    */
-  final lazy val slashPath: String = {
-    val thisOne = "/" + diagnosticDebugName
-    val encElem = enclosingElement
-    if (encElem.isDefined)
-      encElem.get.slashPath + thisOne
-    else
-      thisOne
-  }
+  final lazy val slashPath: String =
+    scPath.filter { _.isInstanceOf[ElementBase] }.map { _.diagnosticDebugName }.mkString("/")
 
   override def toString = diagnosticDebugName
 
   /**
-   * Includes instances. Ie., a global element will appear inside an element ref.
-   * a global group inside a group ref, a global type inside an element or for
-   * derived simple types inside another simple type, etc.
+   * Does not include instances. Ie., walks up lexical nest only.
```

Comment:
Does this mean that diagnostic messages will loose the ability to tell the non-lexical location of errors?

Reply:
Yes. We will have to use the processor's node-stack at runtime to provide the dynamic nest, as we will only have the lexical nest at schema compile time.  We may be able to do better once I have the back reference table in that can provide all the context backpointers where something is being reused. There may be exactly one usage, in which case it is unambiguous.

Reply:
Comment added to code to this effect.